blocks/rich-text/index.js

```diff
@@ -283,11 +283,19 @@ export default class RichText extends Component {
 	 * @param {PasteEvent} event The paste event as triggered by tinyMCE.
 	 */
 	onPaste( event ) {
-		const dataTransfer = event.clipboardData || event.dataTransfer || this.editor.getDoc().dataTransfer;
-		const { items = [], files = [] } = dataTransfer;
+		const dataTransfer =
+			event.clipboardData ||
+			event.dataTransfer ||
+			this.editor.getDoc().dataTransfer ||
+			{ getData: () => '' };
+
+		const { items = [], files = [], types = [] } = dataTransfer;
 		const item = find( [ ...items, ...files ], ( { type } ) => /^image\/(?:jpe?g|png|gif)$/.test( type ) );
+		const plainText = dataTransfer.getData( 'text/plain' );
+		const HTML = dataTransfer.getData( 'text/html' );
 
-		if ( item ) {
+		// Only process file if no HTML is present.
```

Comment:
What about plain text? It's not clear why we'd target HTML specifically, except if we're intending to address Word's behavior alone (in which case, maybe we should comment as such). Otherwise should we be verifying that there are _no other types_ here?

Reply:
This is not necessarily Word specific. Other sources might also add files, in which case we should make sure it's a file without any HTML. Pasted image data might have a URL in the plain text data, so we shouldn't check for it here. I added a comment.

Reply:
Related: #11683